pkg/scheduler/core/generic_scheduler.go

```diff
@@ -903,11 +904,35 @@ func pickOneNodeForPreemption(nodesToVictims map[*v1.Node]*schedulerapi.Victims)
 			lenNodes2++
 		}
 	}
+	if lenNodes2 == 1 {
+		return minNodes2[0]
+	}
+
+	// There are a few nodes with same number of pods.
+	// Find one with the latest start time, i.e. latest of earliest of all pods.
+	theLastestEarliest := util.GetEarliestPodStartTime(nodesToVictims[minNodes2[0]])
+
+	lenNodes1 = 0
+	for i := 0; i < lenNodes2; i++ {
+		node := minNodes2[i]
+
+		earliestPodStartTime := util.GetEarliestPodStartTime(nodesToVictims[node])
+		if earliestPodStartTime.After(theLastestEarliest.Time) {
```

Comment:
`util.GetEarliestPodStartTime` may return nil. You shouldn't dereference return values without checking for nil.